### PR TITLE
Ensure hrtime is monotonic

### LINKED ISSUE #186: Incorrect implementation of hrtime
`hrtime` is not just a function that formats time in a certain way. It must also tick forward at a steady rate, independent of system time. Symfony's implementation relies upon `microtime`, which is at best non-strictly increasing (time sync can pause the system time to sync it, or speed it up), and at worst, may jump backwards in time if the sytem time is changed by a sysadmin, or otherwise.

All this said, it's not obvious to be how to implement this correctly. For reference, PHP's implementation is available at https://github.com/php/php-src/blob/PHP-7.3.9/ext/standard/hrtime.c.

### PR DESCRIPTION
**Pull request by GrahamCampbell**
Closes #186.

### CODE REVIEW

File: src/Php73/Php73.php [hunk 1]
```diff
@@ -32,6 +34,12 @@ public static function hrtime($asNum = false)
         $s = substr($ns, 11) - self::$startAt;
         $ns = 1E9 * (float) $ns;
 
+        if (self::$previousValue !== null) {
```
Comment (nicolas-grekas) [inline src/Php73/Php73.php]: Let's initialize the property at zero and remove this condition

**Comment by GrahamCampbell** [comment]
Hmm, wait this still isn't correct...

File: src/Php73/Php73.php [hunk 2]
```diff
@@ -21,16 +21,28 @@ final class Php73
...
      *
      * @return array|float|int
      */
     public static function hrtime($asNum = false)
     {
-        $ns = microtime(false);
-        $s = substr($ns, 11) - self::$startAt;
-        $ns = 1E9 * (float) $ns;
+        $microtime = microtime(false);
+        $current = 1E9 * (float) $raw;
```
Comment (jdreesen) [inline src/Php73/Php73.php]: What is `$raw`? Should it be `$microtime`?

Comment (GrahamCampbell) [reply src/Php73/Php73.php]: Ah, yeh. I changed my mind about the name of that variable. :P

**Comment by nicolas-grekas** [comment]
Thank you @GrahamCampbell.

**Comment by GrahamCampbell** [comment]
Thanks for finishing this off! :)

**Comment by nicolas-grekas** [comment]
oups, this breaks tests on 7.4, if you wnt to have a look.

**Comment by GrahamCampbell** [comment]
Unrelated, but when I run the tests on my machine, I get a load of warnings before they start:

```
PHP Warning:  Use of undefined constant UUID_TYPE_RANDOM - assumed 'UUID_TYPE_RANDOM' (this will throw an Error in a future version of PHP) in /data/tests/Uuid/UuidTest.php on line 34
PHP Warning:  Use of undefined constant UUID_TYPE_TIME - assumed 'UUID_TYPE_TIME' (this will throw an Error in a future version of PHP) in /data/tests/Uuid/UuidTest.php on line 35
PHP Warning:  Use of undefined constant UUID_TYPE_NULL - assumed 'UUID_TYPE_NULL' (this will throw an Error in a future version of PHP) in /data/tests/Uuid/UuidTest.php on line 122
PHP Warning:  Use of undefined constant UUID_TYPE_RANDOM - assumed 'UUID_TYPE_RANDOM' (this will throw an Error in a future version of PHP) in /data/tests/Uuid/UuidTest.php on line 123
PHP Warning:  Use of undefined constant UUID_TYPE_TIME - assumed 'UUID_TYPE_TIME' (this will throw an Error in a future version of PHP) in /data/tests/Uuid/UuidTest.php on line 124
PHP Warning:  Use of undefined constant UUID_TYPE_TIME - assumed 'UUID_TYPE_TIME' (this will throw an Error in a future version of PHP) in /data/tests/Uuid/UuidTest.php on line 125
PHP Warning:  Use of undefined constant UUID_TYPE_NULL - assumed 'UUID_TYPE_NULL' (this will throw an Error in a future version of PHP) in /data/tests/Uuid/UuidTest.php on line 139
PHP Warning:  Use of undefined constant UUID_VARIANT_DCE - assumed 'UUID_VARIANT_DCE' (this will throw an Error in a future version of PHP) in /data/tests/Uuid/UuidTest.php on line 140
PHP Warning:  Use of undefined constant UUID_VARIANT_DCE - assumed 'UUID_VARIANT_DCE' (this will throw an Error in a future version of PHP) in /data/tests/Uuid/UuidTest.php on line 141
```

**Comment by GrahamCampbell** [comment]
All the tests pass on my PHP 7.2, 7.3, and 7.4. I guess Travis just has an old version.

**Comment by GrahamCampbell** [comment]
```
PHPUnit 7.5.17 by Sebastian Bergmann and contributors.

Runtime:       PHP 7.2.24
Configuration: /data/phpunit.xml.dist

SSS..........................................................   61 / 1206 (  5%)
.............................................................  122 / 1206 ( 10%)
.............................................................  183 / 1206 ( 15%)
.............................................................  244 / 1206 ( 20%)
......................................................SSSSSSS  305 / 1206 ( 25%)
SSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSS  366 / 1206 ( 30%)
SSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSS....  427 / 1206 ( 35%)
..............SSSS................S..........................  488 / 1206 ( 40%)
............SSSSSS..................SSS.......SSSSS..........  549 / 1206 ( 45%)
..................SSSSSSSSSSSSSSSSSSSSS......................  610 / 1206 ( 50%)
.............................................................  671 / 1206 ( 55%)
.............................................................  732 / 1206 ( 60%)
.............................................................  793 / 1206 ( 65%)
.............................................................  854 / 1206 ( 70%)
.............................................................  915 / 1206 ( 75%)
.............................................................  976 / 1206 ( 80%)
................................SSSS......................... 1037 / 1206 ( 85%)
......................................................SSS.... 1098 / 1206 ( 91%)
.........................................SSSSSSSSSSSSS

Time: 1.92 seconds, Memory: 20.00 MB
```

```
PHPUnit 7.5.17 by Sebastian Bergmann and contributors.

Runtime:       PHP 7.3.11
Configuration: /data/phpunit.xml.dist

SSS..........................................................   61 / 1206 (  5%)
.............................................................  122 / 1206 ( 10%)
.............................................................  183 / 1206 ( 15%)
.............................................................  244 / 1206 ( 20%)
......................................................SSSSSSS  305 / 1206 ( 25%)
SSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSS  366 / 1206 ( 30%)
SSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSS....  427 / 1206 ( 35%)
..............SSSS................S..........................  488 / 1206 ( 40%)
............SSSSSS..................SSS......................  549 / 1206 ( 45%)
..................SSSSSSSSSSSSSSSSSSSSS......................  610 / 1206 ( 50%)
.............................................................  671 / 1206 ( 55%)
.............................................................  732 / 1206 ( 60%)
.............................................................  793 / 1206 ( 65%)
.............................................................  854 / 1206 ( 70%)
.............................................................  915 / 1206 ( 75%)
.............................................................  976 / 1206 ( 80%)
................................SSSS......................... 1037 / 1206 ( 85%)
......................................................SSS.... 1098 / 1206 ( 91%)
.........................................SSSSSSSSSSSSS

Time: 2.89 seconds, Memory: 20.00 MB
```

```
PHPUnit 7.5.17 by Sebastian Bergmann and contributors.

Runtime:       PHP 7.4.0RC6
Configuration: /data/phpunit.xml.dist

SSS..........................................................   61 / 1206 (  5%)
.............................................................  122 / 1206 ( 10%)
.............................................................  183 / 1206 ( 15%)
.............................................................  244 / 1206 ( 20%)
......................................................SSSSSSS  305 / 1206 ( 25%)
SSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSS  366 / 1206 ( 30%)
SSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSS....  427 / 1206 ( 35%)
..............SSSS...........................................  488 / 1206 ( 40%)
............SSSSSS..................SSS......................  549 / 1206 ( 45%)
.......................SSSSSSSSSSSSSSSS......................  610 / 1206 ( 50%)
.............................................................  671 / 1206 ( 55%)
.............................................................  732 / 1206 ( 60%)
.............................................................  793 / 1206 ( 65%)
.............................................................  854 / 1206 ( 70%)
.............................................................  915 / 1206 ( 75%)
.............................................................  976 / 1206 ( 80%)
................................SSSS......................... 1037 / 1206 ( 85%)
......................................................SSS.... 1098 / 1206 ( 91%)
.........................................SSSSSSSSSSSSS

Time: 2.82 seconds, Memory: 22.01 MB
```

**Pull request merged**